pkg/services/ngalert/schedule/schedule.go

```diff
@@ -117,33 +109,28 @@ type SchedulerCfg struct {
 }
 
 // NewScheduler returns a new schedule.
-func NewScheduler(cfg SchedulerCfg, appURL *url.URL, stateManager *state.Manager, bus bus.Bus) *schedule {
+func NewScheduler(cfg SchedulerCfg, appURL *url.URL, stateManager *state.Manager /*, bus bus.Bus */) *schedule {
 	ticker := alerting.NewTicker(cfg.C, cfg.Cfg.BaseInterval, cfg.Metrics.Ticker)
 
 	sch := schedule{
-		registry:              alertRuleInfoRegistry{alertRuleInfo: make(map[ngmodels.AlertRuleKey]*alertRuleInfo)},
-		maxAttempts:           cfg.Cfg.MaxAttempts,
-		clock:                 cfg.C,
-		baseInterval:          cfg.Cfg.BaseInterval,
-		log:                   cfg.Logger,
-		ticker:                ticker,
-		evalAppliedFunc:       cfg.EvalAppliedFunc,
-		stopAppliedFunc:       cfg.StopAppliedFunc,
-		evaluator:             cfg.Evaluator,
-		ruleStore:             cfg.RuleStore,
-		instanceStore:         cfg.InstanceStore,
-		metrics:               cfg.Metrics,
-		appURL:                appURL,
-		disableGrafanaFolder:  cfg.Cfg.ReservedLabels.IsReservedLabelDisabled(ngmodels.FolderTitleLabel),
-		stateManager:          stateManager,
-		minRuleInterval:       cfg.Cfg.MinInterval,
-		schedulableAlertRules: alertRulesRegistry{rules: make(map[ngmodels.AlertRuleKey]*ngmodels.AlertRule)},
-		bus:                   bus,
-		alertsSender:          cfg.AlertSender,
+		registry:             alertRuleInfoRegistry{alertRuleInfo: make(map[ngmodels.AlertRuleKey]*alertRuleInfo)},
+		maxAttempts:          cfg.Cfg.MaxAttempts,
+		clock:                cfg.C,
+		baseInterval:         cfg.Cfg.BaseInterval,
+		log:                  cfg.Logger,
+		ticker:               ticker,
+		evalAppliedFunc:      cfg.EvalAppliedFunc,
+		stopAppliedFunc:      cfg.StopAppliedFunc,
+		evaluator:            cfg.Evaluator,
+		ruleStore:            newStoreImpl(cfg.RuleStore, cfg.Metrics, cfg.Logger, !cfg.Cfg.ReservedLabels.IsReservedLabelDisabled(ngmodels.FolderTitleLabel)),
+		instanceStore:        cfg.InstanceStore,
+		metrics:              cfg.Metrics,
+		appURL:               appURL,
+		disableGrafanaFolder: cfg.Cfg.ReservedLabels.IsReservedLabelDisabled(ngmodels.FolderTitleLabel),
+		stateManager:         stateManager,
+		minRuleInterval:      cfg.Cfg.MinInterval,
+		alertsSender:         cfg.AlertSender,
 	}
-
-	bus.AddEventListener(sch.folderUpdateHandler)
```

Comment:
we do not need to listen to events anymore because folders are fetched along with rules